lib/msal-core/src/cache/AuthCache.ts

```diff
@@ -128,16 +131,20 @@ export class AuthCache extends BrowserStorage {// Singleton
     /**
      * Reset all temporary cache items
      */
-    resetTempCacheItems(state: string): void {
+    resetTempCacheItems(state?: string): void {
+        const stateId = state && RequestUtils.parseLibraryState(state).id;
+        const isTokenRenewalInProgress = this.tokenRenewalInProgress(state);
+
         const storage = window[this.cacheLocation];
         let key: string;
         // check state and remove associated cache
-        for (key in storage) {
```

Comment:
This was causing the loop to be run with values from the storage's prototype, e.g. `getItem`, `setItem`. Objects.keys ensure we are just looping the values that are in local storage.